cpp/include/cuvs/cluster/agglomerative.hpp

```diff
@@ -119,6 +120,42 @@ void single_linkage(
   cuvs::cluster::agglomerative::Linkage linkage = cuvs::cluster::agglomerative::Linkage::KNN_GRAPH,
   std::optional<int> c                          = std::make_optional<int>(DEFAULT_CONST_C));
 
+/**
+ * Given a mutual reachability graph and core distances, constructs a linkage over it by computing
+ * the minimum spanning tree and dendrogram. Returns mst edges sorted by weight and the linkage.
+ * Cluster labels are hierarchical and not flattened.
+ * @tparam value_idx
+ * @tparam value_t
+ * @param[in] handle raft handle for resource reuse
+ * @param[in] X data points (size m * n)
+ * @param[in] m number of rows
+ * @param[in] n number of columns
+ * @param[in] metric distance metric to use
+ * @param[in] core_dists core distances (size m)
+ * @param[in] indptr CSR indices of mutual reachability knn graph (size m + 1)
+ * @param[out] mutual_reachability_coo (symmetrized) maximum reachability distance for the k nearest
+ *             neighbors
+ * @param[out] out_mst_src src vertex of MST edges (size m - 1)
+ * @param[out] out_mst_dst dst vertex of MST eges (size m - 1)
+ * @param[out] out_mst_weights weights of MST edges (size m - 1)
+ * @param[out] out_dst children of output
+ * @param[out] out_delta distances of output
+ * @param[out] out_size cluster sizes of output
+ */
+template <typename value_idx, typename value_t, typename nnz_t>
+void build_mutual_reachability_linkage(
+  raft::resources const& handle,
+  raft::device_matrix_view<const value_t, value_idx, raft::row_major> X,
+  cuvs::distance::DistanceType metric,
+  raft::device_vector_view<value_t, value_idx> core_dists,
+  raft::device_vector_view<value_idx, value_idx> indptr,
+  raft::sparse::COO<value_t, value_idx> mutual_reachability_coo,
+  raft::device_vector_view<value_idx, value_idx> out_mst_src,
+  raft::device_vector_view<value_idx, value_idx> out_mst_dst,
+  raft::device_vector_view<value_t, value_idx> out_mst_weights,
+  raft::device_vector_view<value_idx, value_idx> out_children,
+  raft::device_vector_view<value_t, value_idx> out_deltas,
+  raft::device_vector_view<value_idx, value_idx> out_sizes);
```

Comment:
It seems like we are only instantiating this function for `value_t=float` and `value_idx=int`. How about replacing the `value_t` and `value_idx` here with these actual types instead of templates so we don't have to dig down to check which types are supported?